### PR TITLE
Safely handle possible None value encountered in processing and execution of post-job action arguments

### PR DESCRIPTION
**Pull request by hexylena**
Seen in freiburg. Can share all variable values if checking for none is not the right solution.

### CODE REVIEW

**Comment by martenson** [comment]
The name of this PR will help nobody in the 18.01 release notes. ;)

**Comment by hexylena** [comment]
Better? ;)

File: lib/galaxy/jobs/actions/post.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def execute(cls, app, sa_session, action, job, replacement_dict):
         for dataset_assoc in job.output_datasets:
             if action.output_name == '' or dataset_assoc.name == action.output_name:
                 for k, v in action.action_arguments.items():
-                    if v != '':
+                    if v and v != '':
```
Comment (nsoranzo) [inline lib/galaxy/jobs/actions/post.py]: This is equivalent to just:
```python
if v:
```

By looking at the following code it seems that `v` could be a number, so if `v == 0` the following code won't be executed, which seems to be fine to me due to the `if` at line 236, but worth to note.

Comment (hexylena) [reply lib/galaxy/jobs/actions/post.py]: Good point, thanks @nsoranzo

**Review by nsoranzo — APPROVED** [review]

**Comment by dannon** [comment]
@erasche I'd be interested in knowing what the situation was where this was None, but this is definitely a fine, more defensive conditional.

**Comment by bgruening** [comment]
@dannon 
```
Traceback (most recent call last):
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/jobs/runners/__init__.py", line 630, in finish_job
    job_state.job_wrapper.finish(stdout, stderr, exit_code)
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/jobs/__init__.py", line 1350, in finish
    ActionBox.execute(self.app, self.sa_session, pja.post_job_action, job)
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/jobs/actions/post.py", line 416, in execute
    ActionBox.actions[pja.action_type].execute(app, sa_session, pja, job, replacement_dict)
  File "/usr/local/galaxy/galaxy-dist/lib/galaxy/jobs/actions/post.py", line 233, in execute
    if v[0] == 'c':
TypeError: 'NoneType' object has no attribute '__getitem__'
```

**Comment by hexylena** [comment]
utvalg_335

utvalg_336
utvalg_337
utvalg_338
utvalg_339

**Pull request merged**